gui/js/updater.window.js

```diff
@@ -34,22 +34,21 @@ module.exports = class UpdaterWM extends WindowManager {
       log.info({update: info, skipped: this.skipped}, 'Update available')
       // Make sure UI don't show up in front of onboarding after timeout
       if (!this.skipped) {
-        this.skipped = true
-        const shouldUpdate =
-          dialog.showMessageBox({
-            icon: path.resolve(__dirname, '..', 'images', 'icon.png'),
-            title: 'Cozy Drive',
-            message: 'Cozy Drive',
-            detail: translate(
-              'A new version is available, do you want to update?'
-            ),
-            type: 'question',
-            buttons: ['Update', 'Cancel'].map(translate)
-          }) === 0
+        const shouldUpdate = dialog.showMessageBox({
+          icon: path.resolve(__dirname, '..', 'images', 'icon.png'),
+          title: 'Cozy Drive',
+          message: 'Cozy Drive',
+          detail: translate(
+            'A new version is available, do you want to update?'
+          ),
+          type: 'question',
+          buttons: ['Update', 'Cancel'].map(translate)
+        }) === 0
+
         if (shouldUpdate) {
           this.show()
         } else {
-          this.skipped = false
+          this.skipped = true
```

Comment:
@taratatach It is not clear what you want to achieve with this commit:


The previous code will not ask the user to update if `this.skipped` is `false`, and because there is a `setInterval` that call this function, we put `this.skipped` to `true` just before the function displays the message box to the user. If the user says it is ok to update, we show the `electron-updater` update window. If the user don't want to update, we set the flag to `false` because the next time the `setInterval` will call this function we want to display the message box.

Reply:
As I understand it, `this.skipped` should be true when we've skipped an update.
Knowing that I would set it to `true` only when the user has purposefully asked not to update (i.e. the user has skipped the update).

But I understand what the issue would be with the interval. Like I said during the stand-up, if the update process works without my commit, let's revert it.

Reply:
It might be a good thing to rename this variable though, as it can be misleading.

Reply:
As you can see in the current codebase:

We set `this.skipped` to `true` when we are _looking for updates…_:
https://github.com/cozy-labs/cozy-desktop/blob/38bb4ffd777542c3305f22c282c55209196b0177/gui/js/updater.window.js#L79-L92

And its value is used to condition the display of the update window:
https://github.com/cozy-labs/cozy-desktop/blob/38bb4ffd777542c3305f22c282c55209196b0177/gui/js/updater.window.js#L30-L35

So I will drop the commit 43cd97d4c6344d860f36eef30f436eef776f3037.

Reply:
In the first code snippet you pasted above, we skip the update if we reached a timeout when looking for an update. In this situation, we don't want to show the update window since we don't have any update.

This situation is different from the one we're talking about since we're really skipping the update.

When we ask the user if she wants to update or not, we probably don't need to change the value of `this.skipped` since we'll either be:
- not updating and, since its value will be `false` already, we'll show the confirm window again once the check interval is reached
- updating and its value won't matter since we'll stop and relaunch the app entirely

Reply:
`setInterval` will display the message box even if a message box is displayed. That is why we control the display with `this.skipped`.